### PR TITLE
Adding Build Pipeline

### PR DESCRIPTION
**Pull request by KonradSchieban**
Still WIP
 - Adding Cloud Build pipeline steps
 - Rubocop linting
 - InSpec check
 - Dockerfile installs Rubocop
 - Run InSpec against reference environments
 - Store reports in GCS bucket
 - Refactored some controls for Rubocop

### CODE REVIEW

**Comment by aaronlippold** [comment]
Questions:
1. Why tee the json out after the reporter?
2. What is the actual 'threshold' you are tyring to hit? no critical, no high, ... right now it just tests 'does it run' not 'did it run well and as expected'
3. inspec_tools summary --full-json may be cleaner rather than tee to the cli

**Comment by KonradSchieban** [comment]
1. The reference infrastructure we run against is not yet fully compliant. To prevent the pipeline from failing as of now the json is teed out (at least temporarily). As you've mentioned inspec_tools might be cleaner but did not get a chance to play with it yet.
2. This is not yet implemented and really this PR is a first step. Our idea was to test the controls against a "good" (fully compliant) a "bad" (fully noncompliant) and an empty infrastructure (all skipped or partly compliant). Any thoughts on that approach?
3. Agreed, this will be the next step.

File: controls/2.02-logging.rb [hunk 1]
```diff
@@ -50,7 +50,7 @@
   describe "[#{gcp_project_id}] Project level Log sink with an empty filter" do
     subject { empty_filter_sinks }
     it 'is expected to exist' do
-      expect(empty_filter_sinks.count).to be > 0
+      expect(empty_filter_sinks.count).to be.positive
```
Comment (binamov) [inline controls/2.02-logging.rb:53]: I love how this wants you to be positive 😂

Comment (KonradSchieban) [reply controls/2.02-logging.rb]: actually needs to be `be_positive`

**Comment by aaronlippold** [comment]
We just published new docker containers for inspec_tools and heimdall_tools based on ruby apline and I think made them a bit faster.

**Comment by KonradSchieban** [comment]
> We just published new docker containers for inspec_tools and heimdall_tools based on ruby apline and I think made them a bit faster.

Using inspec_tools for compliance check in Cloud Build now (see cloudbuild.yml). Would you suggest we pin the container to a specific version? Currently pulling `latest`. Overall, I think it works well and it's fast.

**Pull request merged**